File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    while(scoville[0] < K):
        if(len(scoville) == 1): # 스코빌 배열의 크기가 1이면 섞는것이 불가능
            return -1
        fNum = heapq.heappop(scoville)
        sNum = heapq.heappop(scoville)
        newNum = fNum + (sNum * 2)
        heapq.heappush(scoville, newNum)
        answer += 1
    return answer

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        
        for(int i = 0 ; i < scoville.length ; i++) {
            minHeap.add(scoville[i]);
        }
        
        boolean flag = true;
        
        while(flag) {
            if(minHeap.peek() >= K) {
                return 0;
            }
            answer += 1;
            int firstMin = minHeap.poll();
            int secondMin = minHeap.poll();   
            int newScoville = firstMin + (secondMin * 2);
            minHeap.add(newScoville);
            if(minHeap.peek() >= K) {
                flag = false;
            }
            else {
                if(minHeap.size() == 1) {
                    return -1;
                }
            }
        }
        
        
        
        return answer;
    }
}